jwst/assign_wcs/nirspec.py

```diff
@@ -539,15 +566,25 @@ def get_open_msa_slits(msa_file, msa_metadata_id, slit_y_range=[-.55, .55]):
 
         # Not allowed....
         else:
-            raise ValueError("MSA configuration file has more than 1 shutter with "
-                             "sources for metadata_id = {}".format(msa_metadata_id))
+            message = ("MSA configuration file has more than 1 shutter with "
+                       "sources for metadata_id = {}".format(msa_metadata_id))
+            log.info(message)
+            raise MSAFileError(message)
 
         # subtract 1 because shutter numbers in the MSA reference file are 1-based.
         shutter_id = xcen + (ycen - 1) * 365
         source_id = slitlets_sid[0]['source_id']
-        source_name, source_alias, stellarity = [
-            (s['source_name'], s['alias'], s['stellarity']) \
-            for s in msa_source if s['source_id'] == source_id][0]
+        try:
+            source_name, source_alias, stellarity = [
+                (s['source_name'], s['alias'], s['stellarity']) \
+                for s in msa_source if s['source_id'] == source_id][0]
+        except IndexError:
+            # all background shutters
+            log.info("All background shutters, slitlet_id is {}".format(slitlet_id))
```

Comment:
Maybe instead of the Yoda-like backwards talking we could say "Slitlet_id n contains all background shutters"